src/crypto/openssl/cose_sign.cpp

```diff
@@ -83,9 +79,83 @@ namespace
 
 namespace ccf::crypto
 {
+  std::optional<int> key_to_cose_alg_id(ccf::crypto::PublicKey_OpenSSL& key)
+  {
+    const auto cid = key.get_curve_id();
+    switch (cid)
+    {
+      case ccf::crypto::CurveID::SECP256K1:
+      case ccf::crypto::CurveID::SECP256R1:
+        std::cout << "Return ES256" << std::endl;
+        return T_COSE_ALGORITHM_ES256;
+      case ccf::crypto::CurveID::SECP384R1:
+        std::cout << "Return ES384" << std::endl;
+        return T_COSE_ALGORITHM_ES384;
+      default:
+        return std::nullopt;
+    }
+  }
+
+  COSEParametersFactory cose_params_int_int(int64_t key, int64_t value)
+  {
+    const size_t args_size = sizeof(key) + sizeof(value);
+    return COSEParametersFactory(
+      [=](QCBOREncodeContext* ctx) {
+        QCBOREncode_AddInt64ToMapN(ctx, key, value);
+      },
+      args_size);
+  }
+
+  COSEParametersFactory cose_params_int_string(
+    int64_t key, std::string_view value)
```

Comment:
I think this needs to be a const string ref, to keep ASAN happy, otherwise we end up with a factory that holds a pointer that comes from an out-of-scope string_view, which is technically still fine because it was assigned from a longer-lived string in a parent scope, but is unnecessarily suspicious.